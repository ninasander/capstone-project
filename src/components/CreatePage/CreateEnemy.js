import React, { useEffect } from 'react'
import { useForm } from 'react-hook-form'
import PageButton from '../Buttons/PageButton'
import styled from 'styled-components/macro'

export default function CreateEnemy({
  addCreatureEntry,
  editCreatureEntry,
  editCreature,
}) {
  const { register, handleSubmit, errors, setValue } = useForm()

  const onSubmit = (creatureEntry, event) => {
    event.target.reset()
    if (editCreature) {
      editCreatureEntry({ ...creatureEntry, _id: editCreature._id })
    } else {
      addCreatureEntry(creatureEntry)
    }
  }

  useEffect(() => {
    if (editCreature) {
      setValue('enemyName', editCreature.enemyName)
      setValue('HP', editCreature.HP)
      setValue('armorClass', editCreature.armorClass)
      setValue('initiative', editCreature.initiative)
    }
  }, [editCreature, setValue])

  return (
    <FormStyled onSubmit={handleSubmit(onSubmit)}>
      <h1>
        {editCreature?.enemyName
          ? 'Edit your enemy creature:'
          : 'Add your enemy creatures:'}
      </h1>
      <FormContainerStyled>
        <section>
          <label htmlFor="enemyName">Enemy name:</label>
          <input
            type="text"
            placeholder="Big Bad Evil Guy"
            name="enemyName"
            ref={register({
              required: true,
              min: 1,
              maxLength: 60,
              pattern: /.*\S.*/,
            })}
          />
          {errors.enemyName && (
            <ErrorInfoStyled>
              Please enter a name (At least 1 character)
            </ErrorInfoStyled>
          )}
        </section>
        <section>
          <label htmlFor="HP">HP:</label>
          <input
            type="number"
            placeholder="Max. Hit Points"
            name="HP"
            ref={register({
              required: true,
              min: 1,
              maxLength: 3,
              pattern: /^[0-9]+$/i,
            })}
          />
          {errors.HP && (
            <ErrorInfoStyled>
              Please enter a positive number (Max. 3 digits)
            </ErrorInfoStyled>
          )}
          <label htmlFor="armorClass">AC:</label>
          <input
            type="number"
            placeholder="Armor Class"
            name="armorClass"
            ref={register({
              required: true,
              min: 0,
              maxLength: 2,
              pattern: /^[0-9]+$/i,
            })}
          />
          {errors.armorClass && (
            <ErrorInfoStyled>
              Please enter a positive number (Max. 2 digits)
            </ErrorInfoStyled>
          )}
        </section>
        <label htmlFor="initiative">Initiative:</label>
        <input
          type="number"
          placeholder="(D20 + Dex)"
          name="initiative"
          ref={register({
            required: true,
            min: 0,
            maxLength: 2,
            pattern: /^[0-9]+$/i,
          })}
        />
        {errors.initiative && (
          <ErrorInfoStyled>
            Please enter a positive number (Max. 2 digits)
          </ErrorInfoStyled>
        )}
        <PageButton
          type="submit"
          buttonText={editCreature?.enemyName ? 'Save Enemy' : 'Add Enemy'}
        />
      </FormContainerStyled>
    </FormStyled>
  )
}

const FormStyled = styled.form`
  color: white;
  h1 {
    font-size: 120%;
  }
  section {
    margin-bottom: 10px;
  }
  label {
    margin-right: 10px;
    font-size: 110%;
  }
  input {
    font-size: 65%;
    text-align: center;
    &::-webkit-outer-spin-button,
    &::-webkit-inner-spin-button {
      -webkit-appearance: none;
      margin: 0;
    }
    &[name='enemyName'] {
      width: 170px;
    }
    &[name='HP'] {
      width: 100px;
      margin-right: 25px;
      -moz-appearance: textfield;
    }
    &[name='armorClass'] {
      width: 90px;
      -moz-appearance: textfield;
    }
    &[name='initiative'] {
      width: 90px;
      margin-right: 10px;
      -moz-appearance: textfield;
    }
  }
`
const FormContainerStyled = styled.div`
  padding: 20px;
  background: hsla(208, 40%, 28%, 0.5);
  border-radius: 5px;
`
const ErrorInfoStyled = styled.p`
  font-size: 75%;
`
