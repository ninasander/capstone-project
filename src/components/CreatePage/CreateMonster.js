import React from 'react'
import { useForm } from 'react-hook-form'

export default function CreateMonster({ onAddCreature }) {
  const { register, handleSubmit } = useForm()
  const onSubmit = (enemyEntry, event) => {
    //const onSubmit = (journalEntry, event)
    event.target.reset()
    onAddCreature(enemyEntry) // onSave(journalEntry)
  }

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <h1>Add your enemy creatures:</h1>
      <label htmlFor="enemyName">Enemy name:</label>
      <input
        type="text"
        placeholder="Enemy name"
        name="enemyName"
        ref={register({ required: true, min: 1, maxLength: 60 })}
      />
      <label htmlFor="HP">HP:</label>
      <input
        type="number"
        placeholder="HP"
        name="HP"
        ref={register({
          required: true,
          min: 1,
          maxLength: 100,
          pattern: /^[0-9]+$/i,
        })}
      />
      <label htmlFor="armorClass">AC:</label>
      <input
        type="number"
        placeholder="Armor Class"
        name="armorClass"
        ref={register({
          required: true,
          min: 0,
          maxLength: 2,
          pattern: /^[0-9]+$/i,
        })}
      />
      <label htmlFor="initiative">Initiative:</label>
      <input
        type="number"
        placeholder="Initiative"
        name="initiative"
        ref={register({
          required: true,
          min: 0,
          maxLength: 2,
          pattern: /^[0-9]+$/i,
        })}
      />

      <button type="submit">Add enemy</button>
    </form>
  )
}
