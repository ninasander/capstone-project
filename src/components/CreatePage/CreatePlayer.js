import React from 'react'
import { useForm } from 'react-hook-form'

export default function CreatePlayer() {
  const { register, handleSubmit } = useForm()
  const onSubmit = (data) => console.log(data)

  return (
    <form onSubmit={handleSubmit(onSubmit)}>
      <h1>Add your players' stats:</h1>
      <label>Enemy name:</label>
      <input
        type="text"
        placeholder="Player name"
        name="Player name"
        ref={register({ required: true, min: 1, maxLength: 60 })}
      />
      <label>AC:</label>
      <input
        type="number"
        placeholder="Armor Class"
        name="Armor Class"
        ref={register({
          required: true,
          min: 0,
          maxLength: 2,
          pattern: /^\S+@\S+$/i,
        })}
      />
      <label>Initiative:</label>
      <input
        type="number"
        placeholder="Initiative"
        name="Initiative"
        ref={register({
          required: true,
          min: 0,
          maxLength: 2,
          pattern: /^[0-9]+$/i,
        })}
      />
      <button type="submit">Add Player</button>
    </form>
  )
}
