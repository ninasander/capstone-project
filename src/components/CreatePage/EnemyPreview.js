import React from 'react'
import styled from 'styled-components/macro'
// import PropTypes from 'prop-types'

// EnemyPreview.propTypes = {
//   enemyName: PropTypes.string.isRequired,
//   armorClass: PropTypes.number.isRequired,
//   HP: PropTypes.number.isRequired,
//   initiative: PropTypes.number.isRequired,
// }

export default function EnemyPreview({
  enemyName,
  armorClass,
  HP,
  initiative,
  creatureEntry,
  handleDelete,
  onEdit,
}) {
  return (
    <>
      <DeleteButtonStyled onClick={handleDeleteClick} id={creatureEntry._id}>
        âœ•
      </DeleteButtonStyled>
      <EnemyEntryStyled>
        <h1>{initiative}</h1>
        <h2>{enemyName}</h2>
        <p>
          HP: {HP}/{HP}
        </p>
        <p>AC: {armorClass}</p>
        <EditButtonStyled onClick={onEdit}>Edit</EditButtonStyled>
      </EnemyEntryStyled>
    </>
  )

  function handleDeleteClick() {
    handleDelete(creatureEntry._id)
  }
}

const DeleteButtonStyled = styled.button`
  color: red;
  margin: 0;
  padding: 0;
  background: none;
  border-style: none;
  font-size: 110%;
  position: relative;
  left: 285px;
  top: 35px;
  z-index: 100;
`
const EditButtonStyled = styled.button`
  color: green;
  background: lightgreen;
  border-style: none;
  grid-column: span 3;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 0;
  width: fit-content;
  justify-self: center;
`

const EnemyEntryStyled = styled.div`
  font-size: 80%;
  margin: 0 20px;
  padding: 20px;
  display: grid;
  grid-gap: 5px 20px;
  justify-content: space-between;
  grid-template-columns: 1fr 2fr 2fr;
  background: hsla(208, 40%, 28%, 1);
  background: linear-gradient(
      180deg,
      hsla(208, 40%, 28%, 1) 0%,
      hsla(208, 40%, 23%, 1) 51%,
      hsla(207, 38%, 18%, 1) 100%
    )
    fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8, 15, 21, 0.3);
  h1 {
    grid-row: span 2;
    align-self: center;
  }
  h2 {
    grid-column: span 2;
    margin: 0;
    font-size: 140%;
    word-break: break-word;
  }
  p {
    margin: 0;
  }
`
