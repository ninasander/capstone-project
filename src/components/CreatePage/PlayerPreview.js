import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import DisplayPlayerAndEnemy from '../EncounterPage/DisplayPlayerAndEnemy'

PlayerPreview.propTypes = {
  creatureEntry: PropTypes.object.isRequired,
  handleDelete: PropTypes.func.isRequired,
  handleEdit: PropTypes.func.isRequired,
}

export default function PlayerPreview({
  handleDelete,
  handleEdit,
  creatureEntry,
}) {
  return (
    <>
      <PlayerEntryStyled>
        <DisplayPlayerAndEnemy
          initiative={creatureEntry.playerInitiative}
          name={creatureEntry.playerName}
          armorClass={creatureEntry.playerArmorClass}
          width="28"
        />
        <ButtonContainer>
          <EditButtonStyled onClick={handleEditClick}>Edit</EditButtonStyled>
          <DeleteButtonStyled
            onClick={handleDeleteClick}
            id={creatureEntry._id}
          >
            Delete
          </DeleteButtonStyled>
        </ButtonContainer>
      </PlayerEntryStyled>
    </>
  )
  function handleDeleteClick() {
    handleDelete(creatureEntry._id)
  }
  function handleEditClick() {
    handleEdit(creatureEntry._id)
  }
}

const PlayerEntryStyled = styled.div`
  margin: 20px;
  padding: 20px;
  display: grid;
  grid-gap: 5px 20px;
  justify-content: space-between;
  grid-template-columns: 1fr 2fr 2fr;
  background: hsla(208, 40%, 28%, 1);
  background: linear-gradient(
      180deg,
      hsla(208, 40%, 28%, 1) 0%,
      hsla(208, 40%, 23%, 1) 51%,
      hsla(207, 38%, 18%, 1) 100%
    )
    fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8, 15, 21, 0.3);

  h2 {
    font-size: 1.5rem;
  }
`
const ButtonContainer = styled.div`
  grid-column: span 3;
  justify-self: center;
`

const DeleteButtonStyled = styled.button`
  color: var(--dark-red);
  background: var(--light-red);
  border-style: none;
  border-radius: 5px;
  padding: 5px 15px;
  margin: 5px 0 0 5px;
  width: fit-content;
`
const EditButtonStyled = styled.button`
  color: var(--dark-green);
  background: var(--light-green);
  border-style: none;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 5px 5px 0 0;
  width: fit-content;
`
