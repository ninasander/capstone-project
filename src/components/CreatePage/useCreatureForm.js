export default function useCreatureForm(addCreatureEntry) {
  // useJournalForm(addEntry)
  const onAddCreature = (entry) => {
    //onSave
    addCreatureEntry(entry) // addEntry(entry)
  }

  return { onAddCreature } //onSave
}
