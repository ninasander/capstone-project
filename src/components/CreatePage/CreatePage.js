import React, { useState } from 'react'
import CreateEnemy from './CreateEnemy'
import CreatePlayer from './CreatePlayer'
import PageButton from '../Buttons/PageButton'
import styled from 'styled-components/macro'
import EnemyPreview from './EnemyPreview'
import PlayerPreview from './PlayerPreview'
import useEnemyPreviewHeight from '../hooks/useEnemyPreviewHeight'
import usePlayerPreviewHeight from '../hooks/usePlayerPreviewHeight'
import { useSpring, animated } from 'react-spring'

export default function CreatePage({
  creatureEntries,
  addCreatureEntry,
  deleteCreatureEntry,
  editCreatureEntry,
}) {
  const [isEnemyPreviewVisible, setIsEnemyPreviewVisible] = useState(false)
  const [isPlayerPreviewVisible, setIsPlayerPreviewVisible] = useState(false)
  const [editCreature, setEditCreature] = useState(undefined)

  const { heightEnemy, bindEnemy } = useEnemyPreviewHeight([])
  const { heightPlayer, bindPlayer } = usePlayerPreviewHeight([])
  const enemyPreviewStyle = {
    ...useSpring({ height: isEnemyPreviewVisible ? heightEnemy : 0 }),
  }
  const playerPreviewStyle = {
    ...useSpring({ height: isPlayerPreviewVisible ? heightPlayer : 0 }),
  }

  const hasEnemy = creatureEntries?.some(
    (creatureEntry) => creatureEntry.enemyName
  )

  const hasPlayer = creatureEntries?.some(
    (creatureEntry) => creatureEntry.playerName
  )

  return (
    <FormsStyled>
      <CreateEnemy
        editCreature={editCreature}
        setEditCreature={setEditCreature}
        editCreatureEntry={editCreatureEntry}
        addCreatureEntry={addCreatureEntry}
      />
      {hasEnemy ? (
        <PageButton
          type="button"
          buttonText={
            isEnemyPreviewVisible
              ? 'Hide Created Enemies ▲'
              : 'Show Created Enemies ▼'
          }
          onClick={toggleEnemyPreview}
        />
      ) : null}
      <PreviewContainerStyled {...bindEnemy} style={enemyPreviewStyle}>
        {creatureEntries.map(
          (creatureEntry) =>
            creatureEntry.enemyName && (
              <EnemyPreview
                creatureEntry={creatureEntry}
                key={creatureEntry._id}
                handleDelete={onDelete}
                handleEdit={onEdit}
              />
            )
        )}
      </PreviewContainerStyled>
      <CreatePlayer
        addCreatureEntry={addCreatureEntry}
        editCreature={editCreature}
        setEditCreature={setEditCreature}
        editCreatureEntry={editCreatureEntry}
      />
      {hasPlayer ? (
        <PageButton
          type="button"
          buttonText={
            isPlayerPreviewVisible
              ? 'Hide Created Players ▲'
              : 'Show Created Players ▼'
          }
          onClick={togglePlayerPreview}
        />
      ) : null}
      <PreviewContainerStyled {...bindPlayer} style={playerPreviewStyle}>
        {creatureEntries.map(
          (creatureEntry) =>
            creatureEntry.playerName && (
              <PlayerPreview
                creatureEntry={creatureEntry}
                key={creatureEntry._id}
                handleDelete={onDelete}
                handleEdit={onEdit}
              />
            )
        )}
      </PreviewContainerStyled>
      {hasEnemy === true && hasPlayer === true ? (
        <a href="/encounter">
          <PageButton type="submit" buttonText="Start Encounter" />
        </a>
      ) : (
        <>
          <PageButton type="submit" buttonText="Start Encounter" />
          <WarningStyled>
            Please add at least one enemy and one player to your encounter.
          </WarningStyled>
        </>
      )}
    </FormsStyled>
  )

  function onDelete(id) {
    const index = creatureEntries.findIndex(
      (creatureEntry) => creatureEntry._id === id
    )
    const entryToDelete = creatureEntries[index]
    deleteCreatureEntry(entryToDelete)
  }
  function onEdit(id) {
    const creatureToEdit = creatureEntries.find(
      (creatureEntry) => creatureEntry._id === id
    )
    setEditCreature(creatureToEdit)
    editCreatureEntry(creatureToEdit)
  }
  function toggleEnemyPreview() {
    setIsEnemyPreviewVisible(!isEnemyPreviewVisible)
  }
  function togglePlayerPreview() {
    setIsPlayerPreviewVisible(!isPlayerPreviewVisible)
  }
}

const PreviewContainerStyled = styled(animated.div)`
  display: block;
  z-index: 200;
  overflow: hidden;
  &::before,
  &::after {
    content: '';
    position: absolute;
    left: 0;
    width: 100%;
    height: 100%;
  }
  &::before {
    height: 0;
    background: linear-gradient(#0002, #0000);
  }
  &::after {
    height: auto;
    bottom: 0;
    background: linear-gradient(#0000, #0001);
  }
`

const FormsStyled = styled.div`
  margin: 20px;
  color: white;
  a {
    text-decoration: none;
  }
`
const WarningStyled = styled.p`
  font-size: 80%;
  text-align: center;
`
