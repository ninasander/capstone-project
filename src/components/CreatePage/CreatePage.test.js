import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import '@testing-library/jest-dom'
import 'mutationobserver-shim'
import CreatePage from './CreatePage'

describe('CreatePage', () => {
  it('renders correctly', () => {
    const creatureEntries = [
      {
        enemyName: 'Enemy 1',
        HP: 50,
        armorClass: 17,
        initiative: 20,
        _id: 5,
      },
      {
        playerName: 'Player 1',
        playerArmorClass: 17,
        playerInitiative: 20,
        _id: 10,
      },
    ]
    const tree = renderer.create(
      <CreatePage creatureEntries={creatureEntries} />
    )
    expect(tree).toMatchSnapshot()
  })
})
