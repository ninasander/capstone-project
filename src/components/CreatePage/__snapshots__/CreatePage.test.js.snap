// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePage renders correctly 1`] = `
.c3 {
  margin: 20px auto 0 auto;
  padding: 10px 20px;
  font-size: 100%;
  font-family: 'Gentium Book Basic',serif;
  display: block;
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-style: none;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

.c1 {
  color: white;
}

.c1 h1 {
  font-size: 120%;
}

.c1 section {
  margin-bottom: 10px;
}

.c1 label {
  margin-right: 10px;
  font-size: 110%;
}

.c1 input {
  font-size: 65%;
  text-align: center;
}

.c1 input::-webkit-outer-spin-button,
.c1 input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c1 input[name='enemyName'] {
  width: 170px;
}

.c1 input[name='HP'] {
  width: 100px;
  margin-right: 25px;
  -moz-appearance: textfield;
}

.c1 input[name='armorClass'] {
  width: 90px;
  -moz-appearance: textfield;
}

.c1 input[name='initiative'] {
  width: 90px;
  margin-right: 10px;
  -moz-appearance: textfield;
}

.c2 {
  padding: 20px;
  background: hsla(208,40%,28%,0.5);
  border-radius: 5px;
}

.c8 {
  color: white;
}

.c8 h1 {
  font-size: 120%;
}

.c8 section {
  margin-bottom: 10px;
}

.c8 label {
  margin-right: 10px;
  font-size: 110%;
}

.c8 input {
  font-size: 65%;
  text-align: center;
}

.c8 input::-webkit-outer-spin-button,
.c8 input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

.c8 input[name='playerName'] {
  width: 175px;
}

.c8 input[name='playerArmorClass'] {
  width: 80px;
  margin-right: 10px;
  -moz-appearance: textfield;
}

.c8 input[name='playerInitiative'] {
  width: 90px;
  margin-right: 10px;
  -moz-appearance: textfield;
}

.c9 {
  padding: 20px;
  background: hsla(208,40%,28%,0.5);
  border-radius: 5px;
}

.c5 {
  color: red;
  margin: 0;
  padding: 0;
  background: none;
  border-style: none;
  font-size: 110%;
  position: relative;
  left: 285px;
  top: 35px;
  z-index: 100;
}

.c7 {
  color: green;
  background: lightgreen;
  border-style: none;
  grid-column: span 3;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  justify-self: center;
}

.c6 {
  font-size: 80%;
  margin: 0 20px;
  padding: 20px;
  display: grid;
  grid-gap: 5px 20px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  grid-template-columns: 1fr 2fr 2fr;
  background: hsla(208,40%,28%,1);
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

.c6 h1 {
  grid-row: span 2;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c6 h2 {
  grid-column: span 2;
  margin: 0;
  font-size: 140%;
  word-break: break-word;
}

.c6 p {
  margin: 0;
}

.c10 {
  color: red;
  margin: 0;
  padding: 0;
  background: none;
  border-style: none;
  font-size: 110%;
  position: relative;
  left: 285px;
  top: 35px;
  z-index: 100;
}

.c12 {
  color: green;
  background: lightgreen;
  border-style: none;
  grid-column: span 3;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  justify-self: center;
}

.c11 {
  font-size: 80%;
  margin: 0 20px;
  padding: 20px;
  display: grid;
  grid-template-columns: 1fr 3fr;
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

.c11 h1 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 10px 0;
}

.c11 h2 {
  margin: 0;
  font-size: 140%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c11 p {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0 20px;
}

.c4 {
  display: block;
  z-index: 200;
  overflow: hidden;
}

.c4::before,
.c4::after {
  content: '';
  position: absolute;
  left: 0;
  width: 100%;
  height: 100%;
}

.c4::before {
  height: 0;
  background: linear-gradient(#0002,#0000);
}

.c4::after {
  height: auto;
  bottom: 0;
  background: linear-gradient(#0000,#0001);
}

.c0 {
  margin: 20px;
  color: white;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  className="c0"
>
  <form
    className="c1"
    onSubmit={[Function]}
  >
    <h1>
      Add your enemy creatures:
    </h1>
    <div
      className="c2"
    >
      <section>
        <label
          htmlFor="enemyName"
        >
          Enemy name:
        </label>
        <input
          name="enemyName"
          placeholder="Big Bad Evil Guy"
          type="text"
        />
      </section>
      <section>
        <label
          htmlFor="HP"
        >
          HP:
        </label>
        <input
          name="HP"
          placeholder="Max. Hit Points"
          type="number"
        />
        <label
          htmlFor="armorClass"
        >
          AC:
        </label>
        <input
          name="armorClass"
          placeholder="Armor Class"
          type="number"
        />
      </section>
      <label
        htmlFor="initiative"
      >
        Initiative:
      </label>
      <input
        name="initiative"
        placeholder="(D20 + Dex)"
        type="number"
      />
      <button
        className="c3"
      >
        Add Enemy
      </button>
    </div>
  </form>
  <button
    className="c3"
    onClick={[Function]}
  >
    Show Created Enemies ▼
  </button>
  <div
    className="c4"
    style={
      Object {
        "height": 0,
      }
    }
  >
    <button
      className="c5"
      id={5}
      onClick={[Function]}
    >
      ✕
    </button>
    <div
      className="c6"
    >
      <h1>
        20
      </h1>
      <h2>
        Enemy 1
      </h2>
      <p>
        HP: 
        50
        /
        50
      </p>
      <p>
        AC: 
        17
      </p>
      <button
        className="c7"
        onClick={[Function]}
      >
        Edit
      </button>
    </div>
  </div>
  <form
    className="c8"
    onSubmit={[Function]}
  >
    <h1>
      Add your players stats:
    </h1>
    <div
      className="c9"
    >
      <section>
        <label>
          Player name:
        </label>
        <input
          name="playerName"
          placeholder="Ser Hero Fancypants"
          type="text"
        />
      </section>
      <section>
        <label>
          AC:
        </label>
        <input
          name="playerArmorClass"
          placeholder="Armor Class"
          type="number"
        />
      </section>
      <label>
        Initiative:
      </label>
      <input
        name="playerInitiative"
        placeholder="(D20 + Dex)"
        type="number"
      />
      <button
        className="c3"
      >
        Add Player
      </button>
    </div>
  </form>
  <button
    className="c3"
    onClick={[Function]}
  >
    Show Created Players ▼
  </button>
  <div
    className="c4"
    style={
      Object {
        "height": 0,
      }
    }
  >
    <div>
      <button
        className="c10"
        id={10}
        onClick={[Function]}
      >
        ✕
      </button>
      <div
        className="c11"
      >
        <h1>
          20
        </h1>
        <h2>
          Player 1
        </h2>
        <p>
          AC: 
          17
        </p>
        <button
          className="c12"
          onClick={[Function]}
        >
          Edit
        </button>
      </div>
    </div>
  </div>
  <a
    href="/encounter"
  >
    <button
      className="c3"
    >
      Start Encounter
    </button>
  </a>
</div>
`;
